Nil pointer de-reference using version v1.3.1

**Describe the bug**
Nil pointer de-reference when trying to access packaged assets in version `v1.3.1`.

**To Reproduce**
Check [`gonut` branch `bump/pina-golada-version`](https://github.com/homeport/gonut/tree/bump/pina-golada-version) to reproduce the problem.

**Expected behavior**
Same as the older version, package and extract work without issues.

**Screenshots**
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x70 pc=0x114923e]

goroutine 1 [running]:
github.com/homeport/pina-golada/pkg/files.WriteToDisk(0x0, 0x0, 0xc0001920c0, 0x3f, 0xc0001a0001, 0xc00019c080, 0x1259df6)
	/Users/mdiester/go/pkg/mod/github.com/homeport/pina-golada@v1.3.1/pkg/files/files_util.go:127 +0x16e
```

--- a/internal/golada/builder/builder.go
+++ b/internal/golada/builder/builder.go
@@ -149,6 +149,7 @@ func (b Builder) BuildFile() (by []byte, err error) {
 		}
 
 		directory := files.NewRootDirectory()
+		topLevelDir := directory // Keep a copy of the top level directory anyway to compress against later
 		isDir := false
 
 		if i, e := os.Stat(methodAnnotation.Asset); e == nil {
@@ -164,8 +165,13 @@ func (b Builder) BuildFile() (by []byte, err error) {
 			return nil, e
 		}
 
-		files.WalkFileTree(directory, func(file files.File) {
-			b.logger.Debug("Gray{Debug➤ Found asset file} White{%s}", file.Name().String())
+		files.WalkDirectoryTree(topLevelDir, func(d files.Directory) {
+			b.logger.Debug("Gray{Debug➤ Found asset directory} White{%s} with permission White{%s}",
+				d.Name().String(), d.PermissionSet().String())
+			for _, file := range d.Files() {
+				b.logger.Debug("Gray{Debug➤ Found asset file} White{%s} with permission White{%s}",
+					file.Name().String(), file.PermissionSet().String())
+			}
 		})
 
 		compressorType := compressor.DefaultRegistry.Find(methodAnnotation.Compressor)
@@ -173,7 +179,7 @@ func (b Builder) BuildFile() (by []byte, err error) {
 			return nil, errors.New("could not find compressor for " + methodAnnotation.Compressor)
 		}
 		buffer := &bytes.Buffer{}
-		if err := compressorType.Compress(directory, buffer); err != nil {
+		if err := compressorType.Compress(topLevelDir, buffer); err != nil {
 			return nil, err
 		}
 

--- a/internal/golada/builder/builder.go
+++ b/internal/golada/builder/builder.go
@@ -28,7 +28,9 @@ import (
 	"errors"
 	"fmt"
 	"github.com/homeport/pina-golada/internal/golada/logger"
+	"github.com/homeport/pina-golada/pkg/files/paths"
 	"go/format"
+	"os"
 	"path/filepath"
 	"strings"
 
@@ -147,7 +149,17 @@ func (b Builder) BuildFile() (by []byte, err error) {
 		}
 
 		directory := files.NewRootDirectory()
-		isDir, e := files.LoadFromDiskAndType(directory, methodAnnotation.Asset)
+		isDir := false
+
+		if i, e := os.Stat(methodAnnotation.Asset); e == nil {
+			// Create a sub directory inside if the target asset is a dir.
+			if i.IsDir() {
+				directory = directory.NewDirectory(paths.Of(i.Name()))
+				isDir = true
+			}
+		}
+
+		e := files.LoadFromDisk(directory, methodAnnotation.Asset)
 		if e != nil {
 			return nil, e
 		}

--- a/pkg/files/files_util.go
+++ b/pkg/files/files_util.go
@@ -184,12 +184,18 @@ func writeFileToDisk(file File, directoryPath string, overwrite bool) (e error) 
 
 func writeDirectoryToDisk(directory Directory, directoryPath string, overwrite bool) (e error) {
 	info, e := os.Stat(directoryPath)
+	permissionSet := directory.PermissionSet()
+
 	if e != nil {
 		if !os.IsNotExist(e) {
 			return e
 		}
 
-		if e := os.MkdirAll(directoryPath, directory.PermissionSet()); e != nil {
+		if e := os.MkdirAll(directoryPath, permissionSet); e != nil {
+			return e
+		}
+
+		if info, e = os.Stat(directoryPath); e != nil {
 			return e
 		}
 	}
@@ -198,8 +204,8 @@ func writeDirectoryToDisk(directory Directory, directoryPath string, overwrite b
 		return fmt.Errorf("provided path pointed to file %s", directoryPath)
 	}
 
-	if overwrite && info.Mode() != directory.PermissionSet() { // Overwrite permissions if overwrite is enabled
-		if err := os.Chmod(directoryPath, directory.PermissionSet()); err != nil {
+	if overwrite && info.Mode() != permissionSet { // Overwrite permissions if overwrite is enabled
+		if err := os.Chmod(directoryPath, permissionSet); err != nil {
 			return err
 		}
 	}

--- a/pkg/files/files_util.go
+++ b/pkg/files/files_util.go
@@ -51,17 +51,16 @@ func WalkDirectoryTree(directory Directory, consumer func(d Directory)) {
 
 // LoadFromDisk loads the content of the paths into the directory recursively
 func LoadFromDisk(directory Directory, path string) (e error) {
-	_, e = LoadFromDiskAndType(directory, path)
-	return e
+	return loadFromDisk(directory, path)
 }
 
-// LoadFromDiskAndType loads the content of the paths into the directory recursively.
+// loadFromDisk loads the content of the paths into the directory recursively.
 // This also returns true if the file located under the path is a directory or false
 // if the file at the path is a flat file.
-func LoadFromDiskAndType(directory Directory, path string) (isDir bool, e error) {
+func loadFromDisk(directory Directory, path string) (e error) {
 	info, statError := os.Stat(path)
 	if statError != nil {
-		return false, statError
+		return statError
 	}
 
 	if info.IsDir() {
@@ -69,27 +68,27 @@ func LoadFromDiskAndType(directory Directory, path string) (isDir bool, e error)
 
 		directoryContent, e := ioutil.ReadDir(path)
 		if e != nil {
-			return true, e
+			return e
 		}
 
 		for _, file := range directoryContent {
 			if file.IsDir() {
-				if err := LoadFromDisk(directory.NewDirectory(paths.Of(file.Name())).WithPermission(file.Mode()), filepath.Join(path, file.Name())); err != nil {
-					return true, err
+				if err := loadFromDisk(directory.NewDirectory(paths.Of(file.Name())).WithPermission(file.Mode()), filepath.Join(path, file.Name())); err != nil {
+					return err
 				}
 			} else {
 				if err := readFileInto(directory, filepath.Join(path, file.Name())); err != nil {
-					return true, err
+					return err
 				}
 			}
 		}
-		return true, nil
+		return nil
 	}
 
 	if err := readFileInto(directory, path); err != nil {
-		return false, err
+		return err
 	}
-	return false, nil
+	return nil
 }
 
 func readFileInto(directory Directory, path string) (e error) {

